src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Pim\Api\Model\EmployeePersonalDetailModel;
+use OrangeHRM\Pim\Traits\Service\EmployeeServiceTrait;
+
+class MyInfoAPI extends Endpoint implements ResourceEndpoint
+{
+    use AuthUserTrait;
+    use EmployeeServiceTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $empNumber =  $this->getAuthUser()->getEmpNumber();
+        $employee = $this->getEmployeeService()->getEmployeeByEmpNumber($empNumber);
+        $this->throwRecordNotFoundExceptionIfNotExist($employee, Employee::class);
+
+        return new EndpointResourceResult(EmployeePersonalDetailModel::class, $employee);
```

Comment:
```suggestion
        return new EndpointResourceResult(EmployeeDetailedModel::class, $employee);
```
:thinking: 
or we can go for multi model support,
`default` => EmployeeModel
`detailed` => EmployeeDetailedModel

Reply:
add multi model support